Lib/email/message.py

```diff
@@ -298,6 +299,17 @@ def get_payload(self, i=None, decode=False):
                     except LookupError:
                         payload = bpayload.decode('ascii', 'replace')
             elif decode:
+
+                # 7bit:   This value means that the email's content contains only 7-bit ASCII characters and no special encoding is required.
+                #         It does not expect any non-ASCII characters or binary data
+                # 8bit:   As mentioned earlier, this value means that the content consists of 8-bit characters,
+                #         including non-ASCII characters, and no special encoding is required.
+                # binary: This value is used to indicate that the content is binary and does not require any special encoding.
+                #         It expects the content to consist of 8-bit characters, including non-ASCII characters and binary data
+                # => if one of these encoding is used, no additional decoding is needed
+                if cte in {'7bit', '8bit', 'binary'}:
+                    return payload
```

Comment:
I think this early `return` is breaking some other tests that are expecting a `bytes` like object but get a `str` instead. For example, see https://github.com/python/cpython/actions/runs/5766215350/job/15633719487?pr=107647#step:18:2455